src/ansys/fluent/core/solver/flobject.py

```diff
@@ -456,10 +457,26 @@ def __getattribute__(self, name):
         if name in super().__getattribute__("child_names"):
             if not self.is_active():
                 raise RuntimeError(f"'{self.path}' is currently not active")
-        return super().__getattribute__(name)
+        try:
+            return super().__getattribute__(name)
+        except AttributeError as ex:
+            attributes = [
+                attribute
+                for attribute in self.__dict__.keys()
+                if not attribute.startswith("_")
+            ]
+            raise AttributeError(
+                allowed_name_error_message("Settings objects", name, attributes)
+            ) from ex
 
     def __setattr__(self, name: str, value):
-        return getattr(self, name).set_state(value)
+        try:
+            return getattr(self, name).set_state(value)
+        except BaseException as ex:
+            if hasattr(getattr(self, name), "allowed_values"):
+                values = getattr(self, name).allowed_values()
+                if value not in values:
```

Comment:
```suggestion
            allowed = getattr(getattr(self, name), "allowed_values", None):
            if allowed and value not in allowed():
```

Reply:
Done. Thank you.

Reply:
@seanpearsonuk 

above change gives following error - 

![image](https://user-images.githubusercontent.com/106588300/220849976-30bd22fd-a987-4675-a3e5-cf7ef7ea518f.png)

So we changed it as follows -

![image](https://user-images.githubusercontent.com/106588300/220850112-31dc1599-62d3-46d3-b985-6cf9f5f90e68.png)